zimui/src/stores/main.ts

```diff
@@ -1,39 +1,52 @@
 import { defineStore } from 'pinia'
 import axios from 'axios'
+import Channel from '@/types/Channel'
+import Topic from '@/types/Topic'
+
+export type RootState = {
+  channelData: Channel | null
+  isLoading: boolean
+  errorMessage: string
+}
 export const useMainStore = defineStore('main', {
-  state: () => ({
-    channel_data: null,
-    is_loading: false,
-    error: '',
-  }),
+  state: () =>
+    ({
+      channelData: null,
+      isLoading: false,
+      errorMessage: '',
+    }) as RootState,
   getters: {},
   actions: {
     async fetchChannel() {
-      this.is_loading = true
-      this.error = ''
-      try {
-        const data = await axios.get('./channel.json')
-        this.channel_data = data.data
-      } catch (error) {
-        this.error = 'Failed to load channel data'
-        this.channel_data = null
-        return false
-      }
-      this.is_loading = false
-      return true
+      this.isLoading = true
+      this.errorMessage = ''
+      return axios.get('./channel.json').then(
+        (response) => {
+          this.isLoading = false
+          this.channelData = response.data as Channel
+        },
+        (_) => {
+          this.isLoading = false
```

Comment:
Maybe log the error in the console?

Reply:
This will be tackled more appropriately by https://github.com/openzim/kolibri/issues/70